packages/base-driver/lib/wd-proxy/proxy.js

```diff
@@ -0,0 +1,404 @@
+              ? `Got response with status ${e.response.status}: ${error}`
+              : `Got response with unknown status: ${error}`
+          );
+        }
+      } else {
+        proxyErrorMsg = `Could not proxy command to the remote server. Original error: ${e.message}`;
+        if (COMPACT_ERROR_PATTERNS.some((p) => p.test(e.message))) {
+          this.log.info(e.message);
+        } else {
+          this.log.info(e.stack);
+        }
+      }
+      throw new errors.ProxyRequestError(proxyErrorMsg, e.response?.data, e.response?.status);
+    }
+  }
+
+  /**
+   *
+   * @param {string} url
+   * @param {import('@appium/types').HTTPMethod} method
+   * @returns {string|undefined}
+   */
+  requestToCommandName(url, method) {
+    /**
+     *
+     * @param {RegExp} pattern
+     * @returns {string|undefined}
+     */
+    const extractCommandName = (pattern) => {
+      const pathMatch = pattern.exec(url);
+      if (pathMatch) {
+        return routeToCommandName(pathMatch[1], method, this.reqBasePath);
+      }
+    };
+    let commandName = routeToCommandName(url, method, this.reqBasePath);
+    if (!commandName && _.includes(url, `${this.reqBasePath}/session/`)) {
+      commandName = extractCommandName(
+        new RegExp(`${_.escapeRegExp(this.reqBasePath)}/session/[^/]+(.+)`)
+      );
+    }
+    if (!commandName && _.includes(url, this.reqBasePath)) {
+      commandName = extractCommandName(new RegExp(`${_.escapeRegExp(this.reqBasePath)}(/.+)`));
+    }
+    return commandName;
+  }
+
+  /**
+   * @param {string} url
+   * @param {import('@appium/types').HTTPMethod} method
+   */
+  async proxyCommand(url, method, body = null) {
+    const commandName = this.requestToCommandName(url, method);
+    if (commandName) {
+      this.log.debug(`Matched '${url}' to command name '${commandName}'`);
+    }
+    return await this.proxy(url, method, body);
+  }
+
+  /**
+   * @param {string} url
+   * @param {import('@appium/types').HTTPMethod} method
+   */
+  async command(url, method, body = null) {
+    let status, data;
+    try {
+      ({status, data} = await this.proxyCommand(url, method, body));
+    } catch (err) {
+      if (isErrorType(err, errors.ProxyRequestError)) {
+        throw err.getActualError();
+      }
+      throw new errors.UnknownError(err.message);
+    }
+    if (status < 300) {
+      return data.value;
+    }
+    if (_.isPlainObject(data.value) && data.value.error) {
+      throw errorFromW3CJsonCode(data.value.error, data.value.message, data.value.stacktrace);
+    }
+    throw new errors.UnknownError(
+      `Did not know what to do with response code '${status}' ` +
+        `and response body '${_.truncate(JSON.stringify(data), {
+          length: 300,
+        })}'`
+    );
+  }
+
+  /**
+   * @param {string} url
+   */
+  getSessionIdFromUrl(url) {
+    const match = url.match(/\/session\/([^/]+)/);
+    return match ? match[1] : null;
+  }
+
+  /**
+   * @param {any} req
```

Comment:
these types are not any. These are Request and Response types by Node Express server